transcendence-app/src/user/user.controller.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common';
+import { UserService } from './user.service';
+import { UserDto } from './dto/user.dto';
+
+@Controller('user')
+export class UserController {
+  constructor(private userService: UserService) {}
+
+  @Post(':id')
+  async addUser(@Param('id') param: string): Promise<UserDto> {
+    const id = Number(param);
+    const user: UserDto = {
+      id: id,
+      provider: 'ft_transcendence',
+      image_url: 'www.img_url.com/hello.jpg',
+      username: `user_${id}`,
+      email: 'afgv@github.com',
+    };
```

Comment:
What is the point of hard-coding all this data? if they are some mocked values, where do you expect they come from?

Reply:
same thing. You have assigned yourself to all of the PRs that implement the [addition of users](https://github.com/42AGV/ft_transcendence/issues/25). What do I do? Sit on my hands?
Just change it when you have a comprehensive solution.

Reply:
Didn't had the context of why you were adding this method and was just questioning it in general, as a new controller added to our system. Now that I have this info, let me explain myself properly :smiley:

An endpoint to add user data to our system expects the client to provide this information. A method like this one is about adding a new user that don't exist yet in our system, this data won't come from the DB. So no need to have anything mocked at this point despite not having any DB implementation ready yet. That is the reason I was asking _what was the point of hard-coding the data at this point_ and also asking _where did you expected this data to come from_.

That user data would have to come from the client and passed, for example, as a param to the controller method, something like:

```ts
  async addUser(@Param('user') param: UserDto): Promise<UserDto> {...}
```
Mocking that UserDto information at this point makes no sense because we are not mocking anything but hardcoding what should come from client.

We should mock this inside the tests (which is the desirable place to mock things as much as we can do it) Something like:

```ts
    await request(server).post({
          id: 123,
          provider: user.provider,
          username: user.provider,
          email: user.email,
          image_url: user.image_url,
        }).expect(HttpStatus.CREATED);
    await request(server).get('/user/1').expect(HttpStatus.OK);
```

That being said, since this method is just  for testing purposes and prone to change I'll just merge it. Just wanted to explain properly my point of view

Reply:
I will do this in the MR I already have open with the user entity.